### PR TITLE
Applied clang-tidy readability-non-const-parameter warning fixes auto…

### PR DESCRIPTION
**Pull request by seanm**
…matically

### CODE REVIEW

**Comment by seanm** [comment]
@qkoziol perhaps this is a step forward in the const battle.  This patch was computer generated by running:

`run-clang-tidy.py -checks=-*,readability-non-const-parameter -fix`

The tool seems smart enough to change only what results in something that compiles. :)

**Review by byrnHDF — APPROVED** [review]
Assuming source formatting fixed.

**Comment by gnuoyd** [comment]
This looks like a step in the right direction.

File: test/direct_chunk.c [hunk 1]
```diff
@@ -76,9 +76,9 @@
 
 /* Local prototypes for filter functions */
 static size_t filter_bogus1(unsigned int flags, size_t cd_nelmts, const unsigned int *cd_values,
-                            size_t nbytes, size_t *buf_size, void **buf);
+                            size_t nbytes, const size_t *buf_size, void **buf);
 static size_t filter_bogus2(unsigned int flags, size_t cd_nelmts, const unsigned int *cd_values,
-                            size_t nbytes, size_t *buf_size, void **buf);
+                            size_t nbytes, const size_t *buf_size, void **buf);
```
Comment (qkoziol) [inline test/direct_chunk.c]: These 2 may not work, since they are supposed to match the signature for H5Z_func_t

File: test/dsets.c [hunk 2]
```diff
@@ -307,7 +307,7 @@ static size_t filter_bogus(unsigned int flags, size_t cd_nelmts, const unsigned
 static htri_t can_apply_bogus(hid_t dcpl_id, hid_t type_id, hid_t space_id);
 static herr_t set_local_bogus2(hid_t dcpl_id, hid_t type_id, hid_t space_id);
 static size_t filter_bogus2(unsigned int flags, size_t cd_nelmts, const unsigned int *cd_values,
-                            size_t nbytes, size_t *buf_size, void **buf);
+                            size_t nbytes, const size_t *buf_size, void **buf);
```
Comment (qkoziol) [inline test/dsets.c]: Same here (H5Z_func_t)

File: tools/test/h5dump/h5dumpgentest.c [hunk 3]
```diff
@@ -150,7 +150,7 @@ const H5Z_class2_t H5Z_MYFILTER[1] = {{
 #define MULTIPLIER          3
 
 static size_t H5Z_filter_dynlibud(unsigned int flags, size_t cd_nelmts, const unsigned int *cd_values,
-                                  size_t nbytes, size_t *buf_size, void **buf);
+                                  size_t nbytes, const size_t *buf_size, void **buf);
```
Comment (qkoziol) [inline tools/test/h5dump/h5dumpgentest.c]: Same here (H5Z_func_t)

**Review by qkoziol — APPROVED** [review]
Aside from the H5Z_func_t issues, this looks good.

**Comment by seanm** [comment]
@qkoziol good catch. see 3rd commit for followup to automated changes.

**Comment by qkoziol** [comment]
> @qkoziol good catch. see 3rd commit for followup to automated changes.

The compiler is OK with this?   The H5Z_class2_t's 'filter' function pointer will be off slightly, since the buf_size pointer is const for these functions, but the H5Z_func_t typedef is non-const.

**Comment by seanm** [comment]
I changed the `H5Z_func_t` typedef to const too.

It compiles for me.

**Comment by qkoziol** [comment]
> I changed the `H5Z_func_t` typedef to const too.
> 
> It compiles for me.

Ah, missed that.  Unfortunately, that typedef can't change, since it's part of the public API for I/O filters.

**Comment by seanm** [comment]
I'm not looking to change this in 1.10 or other stable branches.  Can there never ever be public API changes ever?  :(  This is about as minor an API change as you can get too.

**Comment by qkoziol** [comment]
> I'm not looking to change this in 1.10 or other stable branches. Can there never ever be public API changes ever? :( This is about as minor an API change as you can get too.

Sure the public API can be changed.  :-)   In this case though, we shouldn't change it, since the buf_size pointer is an in/out value.   The filter will update it with the new size of the buffer, after it's been compressed (or otherwise filtered).

**Comment by seanm** [comment]
Oh!  Odd that it compiles then.

**Comment by seanm** [comment]
@qkoziol ok, I think I fixed all the `H5Z_func_t` cases.

**Review by qkoziol — APPROVED** [review]

**Review by derobins — APPROVED** [review]

**Pull request merged**